crates/client-api/src/auth.rs

```diff
@@ -1,11 +1,10 @@
-use std::fmt::Write;
 use std::time::Duration;
 
-use axum::extract::Query;
+use axum::extract::{Query, Request, State};
```

Comment:
on re-reading, I have a question - could most of the refactoring changes in this file be made in a separate PR? Aside from adding `auth_middleware`, it seems like these changes are backwards-compatible, largely independent, and I think they're maybe easier to review if they're separate from the auth functionality changes.

Reply:
I could, if you feel strongly about it, though one could consider them all part and parcel of making the auth code generally more straightforward/readable.

Reply:
(It might be difficult to split them at this point, but yeah if you feel strongly I could try)